### PR TITLE
Make code clang-10 ready

### PR DESCRIPTION
**Pull request by mrks**
I locally installed clang-10 on one of my laptops. Mostly, clang now warns about implicit int to float conversions. While adding static_casts makes the code more verbose, I think that this might make people more aware of potentially lossy casts.

clang-10 support is not yet part of CI, so for now, this is a one time change with (hopefully) no visible effects.

### CODE REVIEW

File: src/lib/strong_typedef.hpp [hunk 1]
```diff
@@ -20,7 +20,7 @@
 
 #define STRONG_TYPEDEF(T, D)                                                                                      \
   namespace opossum {                                                                                             \
-  struct D : boost::totally_ordered1<D, boost::totally_ordered2<D, T>> {                                          \
+  struct D : boost::totally_ordered1<D> {                                                                         \
```
Comment (Bouncner) [inline src/lib/strong_typedef.hpp:23]: Can you explain to me why you changed that?

File: src/test/server/server_test_runner.cpp [hunk 2]
```diff
@@ -431,7 +431,7 @@ TEST_F(ServerTestRunner, TestTransactionConflicts) {
   EXPECT_GE(conflicted_increments, 1);
 
   EXPECT_EQ(successful_increments + conflicted_increments, num_threads * iterations_per_thread);
-  EXPECT_FLOAT_EQ(final_sum - initial_sum, successful_increments);
```
Comment (Bouncner) [inline src/test/server/server_test_runner.cpp:434]: Is this Gtest macro now no longer working with Clang10?

File: src/CMakeLists.txt [hunk 3]
```diff
@@ -41,12 +41,15 @@ endif()
 set(CMAKE_CXX_STANDARD_REQUIRED ON)
 set(CMAKE_CXX_STANDARD 20)
 
+# Hotfix for boost bug (https://github.com/boostorg/asio/issues/312):
+add_compile_options(-DBOOST_ASIO_DISABLE_CONCEPTS)
```
Comment (Bouncner) [inline src/CMakeLists.txt:45]: Is there a way to remind us to remove that option once in a while? Would a TODO help here?

File: src/CMakeLists.txt [hunk 4]
```diff
@@ -41,12 +41,15 @@ endif()
...
+    add_compile_options(-Weverything -Wshadow-all -Wno-c++98-compat -Wno-c++98-compat-pedantic -Wno-documentation -Wno-padded -Wno-global-constructors -Wno-sign-conversion -Wno-exit-time-destructors -Wno-switch-enum -Wno-weak-vtables -Wno-double-promotion -Wno-covered-switch-default -Wno-unused-macros -Wno-newline-eof -Wno-missing-variable-declarations -Wno-weak-template-vtables -Wno-missing-prototypes -Wno-float-equal -Wno-return-std-move-in-c++11 -Wno-unreachable-code-break -Wno-undefined-func-template -Wno-unknown-warning-option -Wno-pass-failed -Wno-ctad-maybe-unsupported -Wno-header-hygiene -Wno-poison-system-directories)
```
Comment (Bouncner) [inline src/CMakeLists.txt:52]: The docs I found mention that this flag is intended for cross-platform builds. Why do we use it? (not that I am against it, sounds useful)

File: src/test/server/server_test_runner.cpp (hunk 2 above)
Comment (mrks) [reply src/test/server/server_test_runner.cpp:434]: No, we just shouldn't compare ints with it ;)

File: src/CMakeLists.txt (hunk 4 above)
Comment (mrks) [reply src/CMakeLists.txt:52]: Didn't really look into it. We don't build cross-platform anyway. My first impression is that fixing it is CMake's job, not ours: https://gitlab.kitware.com/cmake/cmake/issues/17483

Comment (Bouncner) [reply src/CMakeLists.txt:52]: I just wanted to say "nevermind". I thought it's the other way around and we ensure that we allow cross-platform. I misread the doc.

File: src/CMakeLists.txt (hunk 3 above)
Comment (mrks) [reply src/CMakeLists.txt:45]: TODO sounds like something that can be done right now. Given the age of boost versions in Ubuntu, I'd guess that this won't make it into the standard boost any time soon,.

File: src/lib/strong_typedef.hpp (hunk 1 above)
Comment (mrks) [reply src/lib/strong_typedef.hpp:23]: For some reason, clang now thinks that there are two possibilities for `operator==` - the one defined by boost and the one if using the user-defined conversion to the base type. Removing this makes sure that comparisons to the base type always use the second path.

File: src/CMakeLists.txt (hunk 3 above)
Comment (Bouncner) [reply src/CMakeLists.txt:45]: Ack

**Pull request merged**